src/app/modules/learning/task/task.component.ts

```diff
@@ -44,6 +45,32 @@ export class TaskComponent implements OnInit {
     this.setTask();
   }
 
+  public onTabChanged(tabChangeEvent: MatTabChangeEvent): void {
+    if(tabChangeEvent.index === 0) {
+      this.progressService.submissionOpened(this.task.unitId, this.task.id, this.taskProgress.id, this.selectedStep.id)
+      .subscribe();
+    } else if(tabChangeEvent.index === 1) {
+      this.progressService.guidanceOpened(this.task.unitId, this.task.id, this.taskProgress.id, this.selectedStep.id)
+      .subscribe();
+    } else if(tabChangeEvent.index === 2) {
+      this.progressService.exampleOpened(this.task.unitId, this.task.id, this.taskProgress.id, this.selectedStep.id)
```

Comment:
Don't forget that there can be multiple videos at this level. In this file, look at the last function `getNextExample`.

Reply:
onVideoStatusChanged function captures events for any video.